packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -213,8 +214,10 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
 
   @include mdc-theme-dark(".mdc-textfield", true) {
     &:not(.mdc-textfield--focused) {
-      &::after {
-        background-color: $mdc-textfield-error-on-dark;
+      &.mdc-textfield--upgraded {
+        &::after {
```

Comment:
We need to target *both* upgraded and non-upgraded after pseudo elements. If I'm reading this correctly, it will only target the upgraded form.